library/alloc/src/collections/vec_deque.rs

```diff
@@ -982,7 +983,12 @@ impl<T> VecDeque<T> {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn iter_mut(&mut self) -> IterMut<'_, T> {
-        IterMut { tail: self.tail, head: self.head, ring: unsafe { self.buffer_as_mut_slice() } }
+        IterMut {
+            tail: self.tail,
+            head: self.head,
+            ring: unsafe { self.buffer_as_mut_slice() },
```

Comment:
This could be safe via `ptr::slice_from_raw_parts_mut` (that doesn't change anything about the safety, but the alternative is to add a safety comment here explaining why this is fine).

Reply:
Good catch! There is no reason to create a reference here.